osquery/dispatcher/scheduler.cpp

```diff
@@ -150,38 +151,67 @@ inline void launchQuery(const std::string& name, const ScheduledQuery& query) {
 }
 
 void SchedulerRunner::start() {
-  // Start the counter at the second.
-  auto i = osquery::getUnixTime();
-  for (; (timeout_ == 0) || (i <= timeout_); ++i) {
-    Config::get().scheduledQueries(
-        ([&i](const std::string& name, const ScheduledQuery& query) {
-          if (query.splayed_interval > 0 && i % query.splayed_interval == 0) {
-            TablePlugin::kCacheInterval = query.splayed_interval;
-            TablePlugin::kCacheStep = i;
-            launchQuery(name, query);
-          }
-        }));
+  auto tasks = TaskSchedule{};
+  Config::get().scheduledQueries(
+      [&tasks](const std::string& name, const ScheduledQuery& query) {
+        if (query.splayed_interval > 0) {
+          tasks.add(
+              [name, &query](const auto start_time) {
```

Comment:
Did you measure performance here? 
I see number of possible performance problems here, but it completely depends on count of queries. 
1. Converting lambda to std::function maybe expensive here
2. Create new lambda for every query:
 - we will need to allocate/store them 
 - function itself relying  on captured value that of control 
I would suggest do more functional approach and send required data to function as params, and reuse single lambda

Reply:
The conversion from lambda to `std::function` take a place only when new regular query is added. We don't create a lambda for every query run.

Using a `std::function` here give us some advantages. We can use a regular query any kind of function with particular signature.